File: Backend/EdushareBackend/Entities/Dtos/MaterialShortViewDto.cs
using Entities.Models;
using Microsoft.AspNetCore.Identity;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Entities.Dtos
{
    public class MaterialShortViewDto
    {
        public string Id { get; set; } = string.Empty;
        public string Title { get; set; } = string.Empty;
        public AppUserMaterialShortView Uploader { get; set; } = null!;
        public DateTime UploadDate { get; set; }
    }
}

File: Frontend/eduShare/src/app/dtos/material-short-view-dto.ts
import { Subject } from "../models/subject"
import { UploaderDto } from "./uploader-dto"

export interface MaterialShortViewDto {
    id: string
    title: string
    isRecommended: boolean
    isExam: boolean
    subject: Subject
    uploader: UploaderDto
    uploadDate: string
}
